### PR TITLE
Update staging test versions

### PR DESCRIPTION
**Pull request by msau42**

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-staging/controller_images.yaml [hunk 1]
```diff
@@ -7,8 +7,8 @@ spec:
     spec:
       containers:
         - name: csi-provisioner
-          image: gcr.io/gke-release-staging/csi-provisioner:v1.0.0-gke.0
+          image: gcr.io/gke-release-staging/csi-provisioner:v1.0.1-rc1
```
Comment (davidz627) [inline deploy/kubernetes/overlays/prow-gke-staging/controller_images.yaml]: not `v1.0.1-gke.0-rc1`?

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-staging/controller_images.yaml]: there's an issue with our ci config and it's not building the -gke.0 variant for "-rcX" tags

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-gke-staging/controller_images.yaml]: is there a tracking issue?

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-staging/controller_images.yaml]: I don't think we need -gke.x variants for rc's

**Pull request merged**